dhall/src/Dhall.hs

```diff
@@ -195,6 +206,68 @@ instance HasEvaluateSettings InputSettings where
 instance HasEvaluateSettings EvaluateSettings where
   evaluateSettings = id
 
+-- | Parse an expression, using the supplied `InputSettings`
+parseWithSettings :: InputSettings -> Text -> IO (Expr Src Import)
+parseWithSettings settings text = do
+    Core.throws (Dhall.Parser.exprFromText (view sourceName settings) text)
+
+-- | Type-check an expression, using the supplied `InputSettings`
+typecheckWithSettings :: InputSettings -> Expr Src Void -> IO ()
```

Comment:
I'm going to leave this as is for right now (as I'm trying to cut a release soon), and also because a `MonadThrow` constraint alone is not enough.  It would need to also have `MonadIO`, but then you could subsume the `MonadThrow` constraint into the `MonadIO` constraint (by using `liftIO . throwIO`), and then I feel like leaving it as `IO` instead of `MonadIO` is preferable.

Reply:
@Gabriella439 I think `Dhall.Core.throws` adds the necessity for `IO`, not `Dhall.TypeChech.typeWith` or `Dhall.Parser.exprFromText`. While I am really looking forward to a new release, I thought it might be best to change the API before that and not when it is already published and part of the user interface.